test/main.sh

```diff
@@ -273,13 +273,22 @@ if [ -n "${GITHUB_ACTIONS:-}" ]; then
 fi
 
 # allow for running a specific set of tests
-if [ "$#" -gt 0 ] && [ "$1" != "all" ] && [ "$1" != "cluster" ] && [ "$1" != "standalone" ]; then
+if [ "$#" -gt 0 ] && [ "$1" != "all" ] && [ "$1" != "cluster" ] && [ "$1" != "standalone" ] && [ "$1" != "tshell" ]; then
   run_test "test_${1}"
   # shellcheck disable=SC2034
   TEST_RESULT=success
   exit
 fi
 
+if [ "${1:-"all"}" = "tshell" ]; then
```

Comment:
What is this and why do we need it? Commit message is lacking explanation.

Reply:
It gives you an interactive "test shell session". I augmented the commit messages and added comments in the script and `Makefile`.

Reply:
can we call it `make testshell` to be clearer?

Reply:
Naming is hard. It started as `tash` for `test + bash`, then felt that `tshell` was more descriptive while still being short but I guess we go all in with explicit and use `test-shell`. I added the `-` as our other target separate words with it already.